airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/AbstractPostgresSourceDatatypeTest.java

```diff
@@ -0,0 +1,528 @@
+              .addExpectedValues(null, "-32768", "32767")
+              .build());
+    }
+
+    for (final String type : Set.of("smallserial", "serial2")) {
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType(type)
+              .airbyteType(JsonSchemaType.INTEGER)
+              .addInsertValues("1", "32767", "0", "-32767")
+              .addExpectedValues("1", "32767", "0", "-32767")
+              .build());
+    }
+
+    for (final String type : Set.of("serial", "serial4")) {
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType(type)
+              .airbyteType(JsonSchemaType.INTEGER)
+              .addInsertValues("1", "2147483647", "0", "-2147483647")
+              .addExpectedValues("1", "2147483647", "0", "-2147483647")
+              .build());
+    }
+
+    // time without time zone
+    for (final String fullSourceType : Set.of("time", "time without time zone")) {
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType("time")
+              .fullSourceDataType(fullSourceType)
+              .airbyteType(JsonSchemaType.STRING_TIME_WITHOUT_TIMEZONE)
+              // time column will ignore time zone
+              .addInsertValues("null", "'13:00:01'", "'13:00:02+8'", "'13:00:03-8'", "'13:00:04Z'", "'13:00:05.01234Z+8'", "'13:00:00Z-8'")
+              .addExpectedValues(null, "13:00:01.000000", "13:00:02.000000", "13:00:03.000000", "13:00:04.000000", "13:00:05.012340",
+                  "13:00:00.000000")
+              .build());
+    }
+
+    // time with time zone
+    for (final String fullSourceType : Set.of("timetz", "time with time zone")) {
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType("timetz")
+              .fullSourceDataType(fullSourceType)
+              .airbyteType(JsonSchemaType.STRING_TIME_WITH_TIMEZONE)
+              .addInsertValues("null", "'13:00:01'", "'13:00:00+8'", "'13:00:03-8'", "'13:00:04Z'", "'13:00:05.012345Z+8'", "'13:00:06.00000Z-8'")
+              // A time value without time zone will use the time zone set on the database, which is Z-7,
+              // so 13:00:01 is returned as 13:00:01-07.
+              .addExpectedValues(null, "13:00:01.000000-07:00", "13:00:00.000000+08:00", "13:00:03.000000-08:00", "13:00:04.000000Z",
+                  "13:00:05.012345-08:00", "13:00:06.000000+08:00")
+              .build());
+    }
+
+    // timestamp without time zone
+    for (final String fullSourceType : Set.of("timestamp", "timestamp without time zone")) {
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType("timestamp")
+              .fullSourceDataType(fullSourceType)
+              .airbyteType(JsonSchemaType.STRING_TIMESTAMP_WITHOUT_TIMEZONE)
+              .addInsertValues("TIMESTAMP '2004-10-19 10:23:00'", "TIMESTAMP '2004-10-19 10:23:54.123456'", "null")
```

Comment:
let's add a BC timestamp to the timestamp and timestamptz test cases, e.g. `"TIMESTAMP '3001-10-19 10:23:54.123456 BC'"`

Reply:
I actually did this locally and had to make this change to get source-postgres to pass the tests https://github.com/airbytehq/airbyte/pull/14534/commits/fb9d301af86d9bdebcf62e0acca22f18c7134629

Reply:
(also, took the opportunity to reuse the existing method in AbstractJdbcCompatibleSourceOperations)